File: mktrs-spring-boot-native-java-poc/src/main/java/com/themusketeers/sbnative/common/exception/ApiException.java
/*----------------------------------------------------------------------------*/
/* Source File:   APIEXCEPTION.JAVA                                           */
/* Copyright (c), 2023 The Musketeers                                         */
/*----------------------------------------------------------------------------*/
/*-----------------------------------------------------------------------------
 History
 Jun.21/2023  COQ  File created.
 -----------------------------------------------------------------------------*/
package com.themusketeers.sbnative.common.exception;

/**
 * An exception model for errors in downstream service requests.
 *
 * @author COQ - Carlos Adolfo Ortiz Q.
 */
public class ApiException extends RuntimeException {

    private final String url;
    private final String errorCode;

    /**
     * Constructor with parameters.
     *
     * @param message   Description of the exception.
     * @param url       Identifies the location where the exception occurred.
     * @param errorCode HTTP error number for the exception.
     */
    public ApiException(String message, String url, String errorCode) {
        super(message);
        this.url = url;
        this.errorCode = errorCode;
    }
}

File: mktrs-spring-boot-native-kotlin-poc/src/main/kotlin/com/themusketeers/sbnative/common/exception/ApiException.kt
/*----------------------------------------------------------------------------*/
/* Source File:   APIEXCEPTION.KT                                             */
/* Copyright (c), 2023 The Musketeers                                         */
/*----------------------------------------------------------------------------*/
/*-----------------------------------------------------------------------------
 History
 Jun.21/2023  COQ  File created.
 -----------------------------------------------------------------------------*/
package com.themusketeers.sbnative.common.exception

/**
 * An exception model for errors in downstream service requests.
 * @param message   Description of the exception.
 * @param url       Identifies the location where the exception occurred.
 * @param errorCode HTTP error number for the exception.
 * @author COQ - Carlos Adolfo Ortiz Q.
 */
class ApiException(message: String, private val url: String, private val errorCode: String) : RuntimeException(message)
